Permanent/ViewModels/FilesViewModel.swift

```diff
@@ -433,4 +433,39 @@ extension FilesViewModel {
             }
         }
     }
+    
+    func rename(file: FileViewModel, name: String?, then handler: @escaping ServerResponse) {
+        var params: UpdateRecordParams = (name, nil, nil, nil, file.recordId, file.folderLinkId, file.archiveNo)
+        var apiOperation = APIOperation(FilesEndpoint.update(params: params))
+        
+        if file.type.isFolder {
```

Comment:
can we do an if/else type of logic here? looks a bit confusing to set up the params for the file, then overwrite them for the folder

Reply:
Added else condition to if statement.